settings.py.dist

```diff
@@ -10,57 +12,57 @@ ADMINS = (
+DATABASE_NAME = config('DATABASE_NAME', default='master_naturalist_db')             # Or path to database file if sqlite3.
+DATABASE_USER = config('DATABASE_USER', default='root')             # Not used with sqlite3.
+DATABASE_PASSWORD = config('DATABASE_PASSWORD', default='scott')         # Not used with sqlite3.
+DATABASE_HOST = config('DATABASE_HOST', default='')             # Set to empty string for localhost.
+DATABASE_PORT = config('DATABASE_PORT', default='')             # Set to empty string for default.
 
 # prefix used for the site.  ie. http://myhost.com/<SITE_ROOT>/
 # for the django standalone server this should be ""
 # for apache this is the url the site is mapped to, probably /pgd
-SITE_ROOT = ''
+SITE_ROOT = config('SITE_ROOT', default='')
 
 # absolute path to the docroot of this site
-DOC_ROOT = ''
+DOC_ROOT = config('DOC_ROOT', default='')
 
 # Local time zone for this installation. Choices can be found here:
 # http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
 # although not all choices may be available on all operating systems.
 # If running in a Windows environment this must be set to the same as your
 # system time zone.
-TIME_ZONE = 'America/Chicago'
+TIME_ZONE = config('TIME_ZONE', default='America/Chicago')
 
 # Language code for this installation. All choices can be found here:
 # http://www.i18nguy.com/unicode/language-identifiers.html
-LANGUAGE_CODE = 'en-us'
+LANGUAGE_CODE = config('LANGUAGE_CODE', default='en-us')
 
-SITE_ID = 1
+SITE_ID = config('SITE_ID', default=1, cast=int)
 
 #Google analytics ID. Enter the full id, as in: UA-xxxxxx-x 
-GOOGLE_ID = 'UA-9880071-1'
+GOOGLE_ID = config('GOOGLE_ID', default='UA-9880071-1')
 
 # If you set this to False, Django will make some optimizations so as not
 # to load the internationalization machinery.
-USE_I18N = True
+USE_I18N = config('USE_I18N', default=True, cast=bool)
 
 # Absolute path to the directory that holds media.
 # Example: "/home/media/media.lawrence.com/"
-MEDIA_ROOT = '%s/static' % DOC_ROOT
+MEDIA_ROOT = config('MEDIA_ROOT', default='%s/static' % DOC_ROOT)
 
 # URL that handles the media served from MEDIA_ROOT. Make sure to use a
 # trailing slash if there is a path component (optional in other cases).
 # Examples: "http://media.lawrence.com", "http://example.com/media/"
-MEDIA_URL = '%s/%s' % (SITE_ROOT, 'static')
+MEDIA_URL = config('MEDIA_URL', default='%s/%s' % (SITE_ROOT, 'static'))
 
 # URL prefix for admin media -- CSS, JavaScript and images. Make sure to use a
 # trailing slash.
 # Examples: "http://foo.com/media/", "/media/".
-ADMIN_MEDIA_PREFIX = '/media/'
+ADMIN_MEDIA_PREFIX = config('ADMIN_MEDIA_PREFIX', default='/media/')
 
 # Make this unique, and don't share it with anybody.
-SECRET_KEY = 'g3#f517j(b4$ze*3_8pi3i$=1@mk=fnbb_gunjwk_f8whrbo=a'
+SECRET_KEY = config('SECRET_KEY', default='g3#f517j(b4$ze*3_8pi3i$=1@mk=fnbb_gunjwk_f8whrbo=a')
```

Comment:
Generating one is difficult to automate, because it needs to be the same between runs for some apps. It would need to:

- Generate a key
- Write key to file
- Use key from file in subsequent runs